### PR TITLE
Add make-robot-model-from-name function to create instance of robot-model from the name of the robot

### PR DESCRIPTION
**Pull request by garaemon**
Discussion from here https://github.com/jsk-ros-pkg/jsk_visualization/issues/227

* get list of subclasses of robot-model
* find a class whose name is equal to the specified name (checking with `-robot` and `robot-model` suffix)
* instantiate it

### CODE REVIEW

**Comment by garaemon** [comment]
irtutil.l is correct location to define it?

**Comment by snozawa** [comment]
I'd like to know when this function is necessary. 

To generate robot model, 
1. we load robot model
2. we create robot model instance

```
(load "package://pr2eus/pr2.l")
(pr2)
```

Your proposed function is generalize 2, but before 2, we need to load model like 1.
So, you may need to generalize 1. 

Is ``robots.l`` required to be define?

**Comment by garaemon** [comment]
> Your proposed function is generalize 2, but before 2, we need to load model like 1.
> So, you may need to generalize 1.

Eactly, It's not yet a perfect solution.

**Comment by garaemon** [comment]
If we have another rule, we can generalize 2.

For example, all the robot model should provide `:${robot-name}-model` and all the files should be listed in `*load-path*`

If it's acceptable, we can automaticaly call `require`.

**Comment by garaemon** [comment]
like this
```
(defun make-robot-model-from-name* (name &rest args)
  (or (apply #'make-robot-model-from-name name args)
       (progn (require (intern (string-upcase (format nil "~A-model" name)) "KEYWORD"))
                   (apply #'make-robot-model-from-name name args))))
```

File: irteus/irtutil.l [hunk 1]
```diff
@@ -368,6 +368,19 @@
           (reverse ret))
       (close strm))))
 
+(defun make-robot-model-from-name (name &rest args)
+  "make a robot model from string: (make-robot-model \"pr2\")"
+  (let ((name (string-downcase name)))
+    (let ((klasses (flatten (send robot-model :hierarchy))))
+      (dolist (klass klasses)
+        (let ((klass-name (string-downcase (send klass :name))))
+        (if (or (string= klass-name name)
```
Comment (snozawa) [inline irteus/irtutil.l]: I'd like to keep searching rules simple.
Why isn't it sufficient to search [robot-name]-modef classl?
Which robot has class name like [robot-name]-robot-model?

Moreover, currently `pr2-sensor-robot` cannot be searched by this.
However, in this case, we may rename:
- `pr2-robot` -> `pr2-without-sensor-robot` (converted from euscollada)
- `pr2-sensor-robot` -> `pr2-robot` (defined in the last part of pr2.l)

Comment (garaemon) [inline irteus/irtutil.l]: in irteus, robot like `cmu-bvh-robot-model` is defined. I don't mind support only [robot-name]-model or both of [robot-name]-model and [robot-name]-robot-model.

Comment (garaemon) [inline irteus/irtutil.l]: I will update to support only [robot-name]-model

**Comment by garaemon** [comment]
updated, please check it

**Pull request merged**